node/lib/util/read_repo_ast_util.js

```diff
@@ -324,6 +336,35 @@ exports.readRAST = co.wrap(function *(repo) {
                 }
             }
         }
+
+        // Check for changes to submodules not reflected in the index (other
```

Comment:
Yeah, it was getting a little out of control. Moved the chunk of logic for inspecting the workdir and index out.